### PR TITLE
Storage Protocol Refactoring

### PR DESCRIPTION
**Pull request by Delaunay**
Add a new Storage Protocol to make Orion independent from the storage method.

### CODE REVIEW

File: src/orion/storage/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
```
Comment (bouthilx) [inline src/orion/storage/__init__.py]: This file should be deleted. It will break namespace packaging.

File: src/orion/storage/base.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# pylint:disable=protected-access,too-many-public-methods
```
Comment (bouthilx) [inline src/orion/storage/base.py]: Where is the protected access? Can you move `too-many-public-methods` to BaseStorageProtocol.

File: src/orion/storage/legacy.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# pylint:disable=protected-access,too-many-public-methods
```
Comment (bouthilx) [inline src/orion/storage/legacy.py]: idem

**Comment by Delaunay** [comment]
@bouthilx check this out!

File: src/orion/core/worker/producer.py [hunk 4]
```diff
@@ -72,7 +80,10 @@ def produce(self):
                 try:
                     new_trial.parents = self.naive_trials_history.children
                     log.debug("#### Register new trial to database: %s", new_trial)
-                    self.experiment.register_trial(new_trial)
+
+                    self.protocol.create_trial(new_trial)
+                    # self.experiment.register_trial(new_trial)
```
Comment (bouthilx) [inline src/orion/core/worker/producer.py]: leftover

File: src/orion/core/worker/producer.py [hunk 5]
```diff
@@ -96,7 +107,9 @@ def update(self):
     def _update_algorithm(self):
         """Pull newest completed trials to update local model."""
         log.debug("### Fetch trials to observe:")
-        completed_trials = self.experiment.fetch_completed_trials()
+        # completed_trials = self.experiment.fetch_completed_trials()
```
Comment (bouthilx) [inline src/orion/core/worker/producer.py]: leftover

File: src/orion/core/worker/__init__.py [hunk 6]
```diff
@@ -41,8 +41,13 @@ def reserve_trial(experiment, producer):
 
 def workon(experiment, worker_trials=None):
     """Try to find solution to the search problem defined in `experiment`."""
-    producer = Producer(experiment)
-    consumer = Consumer(experiment)
+    # backend = 'debug:'
+    # protocol = StorageProtocol('track', experiment=experiment, uri='file://orion_results.json')
```
Comment (bouthilx) [inline src/orion/core/worker/__init__.py]: leftover

**Comment by bouthilx** [comment]
Now #210

**Pull request closed without merging**